homeassistant/components/proxmoxve/binary_sensor.py

```diff
@@ -1,96 +1,123 @@
-                coordinator = host_name_coordinators[node_name][dev_id]
+    for container_id in config_entry.data[CONF_LXC]:
+        coordinator = coordinators[container_id]
 
-                # unfound case
-                if (coordinator_data := coordinator.data) is None:
-                    continue
+        # unfound container case
+        if coordinator.data is None:
+            continue
 
-                name = coordinator_data["name"]
-                sensor = create_binary_sensor(
-                    coordinator, host_name, node_name, dev_id, name
-                )
-                sensors.append(sensor)
+        container_sensor = create_binary_sensor(
+            coordinator=coordinator,
+            vm_id=container_id,
+            key="status",
+            name="Status",
+            config_entry=config_entry,
+            info_device=device_info(
+                hass=hass,
+                config_entry=config_entry,
+                api_category=ProxmoxType.LXC,
+                vm_id=container_id,
+            ),
+        )
+        sensors.append(container_sensor)
 
-    add_entities(sensors)
+    async_add_entities(sensors)
 
 
-def create_binary_sensor(coordinator, host_name, node_name, vm_id, name):
+def create_binary_sensor(
+    coordinator,
+    vm_id,
+    key,
+    name,
+    config_entry,
+    info_device,
+):
     """Create a binary sensor based on the given data."""
     return ProxmoxBinarySensor(
         coordinator=coordinator,
-        unique_id=f"proxmox_{node_name}_{vm_id}_running",
-        name=f"{node_name}_{name}",
+        unique_id=f"{config_entry.entry_id}_{vm_id}_{key}",
+        name=name,
         icon="",
-        host_name=host_name,
-        node_name=node_name,
+        device_class=BinarySensorDeviceClass.RUNNING,
         vm_id=vm_id,
+        info_device=info_device,
     )
 
 
 class ProxmoxBinarySensor(ProxmoxEntity, BinarySensorEntity):
     """A binary sensor for reading Proxmox VE data."""
 
-    _attr_device_class = BinarySensorDeviceClass.RUNNING
+    _attr_has_entity_name = True
 
     def __init__(
         self,
         coordinator: DataUpdateCoordinator,
         unique_id,
         name,
         icon,
-        host_name,
-        node_name,
+        device_class,
         vm_id,
+        info_device,
     ) -> None:
         """Create the binary sensor for vms or containers."""
-        super().__init__(
-            coordinator, unique_id, name, icon, host_name, node_name, vm_id
-        )
+        super().__init__(coordinator, unique_id, name, icon)
+
+        self._attr_device_class = device_class
+        self._attr_device_info = info_device
 
     @property
-    def is_on(self):
+    def is_on(self) -> None:
         """Return the state of the binary sensor."""
         if (data := self.coordinator.data) is None:
             return None
 
-        return data["status"] == "running"
+        return data[ProxmoxKeyAPIParse.STATUS] == "running"
```

Comment:
This could be done in a preliminary PR

Reply:
I will wait for the #90359 review to do this PR…

Reply:
Included in PR #90371. In draft until #90359 merge.